services/payments/live_solana_test.go

```diff
@@ -0,0 +1,372 @@
+		Authorizations: []paymentLib.PaymentAuthorization{{}, {}, {}},
+	}
+
+	solMachine, mockTransitionHistory, marshaledState := setupState(state, mint, t)
+
+	finalState := driveHappyPathTransitions(
+		ctx,
+		state,
+		mockTransitionHistory,
+		solMachine,
+		marshaledState,
+		t,
+	)
+
+	staticAta, _, err := common.FindAssociatedTokenAddress(
+		common.PublicKeyFromString(state.PaymentDetails.To),
+		common.PublicKeyFromString(mint),
+	)
+	must.Nil(t, err)
+
+	checkTransactionMatchesPaymentDetails(ctx, t, solMachine.solanaRpcEndpoint, staticAta, state, *finalState)
+}
+
+func checkTransactionMatchesPaymentDetails(
+	ctx context.Context,
+	t *testing.T,
+	endpoint string,
+	ata common.PublicKey,
+	state, finalState paymentLib.AuthenticatedPaymentState,
+) {
+	solanaData := chainIdempotencyData{}
+	err := json.Unmarshal(finalState.ExternalIdempotency, &solanaData)
+	must.Nil(t, err)
+
+	solClient := client.NewClient(endpoint)
+	must.Nil(t, err)
+
+	var txn rpc.JsonRpcResponse[*rpc.GetTransaction]
+	t.Log("Fetching transaction data")
+	b58Signature := base58.Encode(solanaData.Transaction.Signatures[0])
+	txn, err = solClient.RpcClient.GetTransactionWithConfig(ctx, b58Signature, rpc.GetTransactionConfig{
+		Encoding:   rpc.TransactionEncodingJsonParsed,
+		Commitment: rpc.CommitmentConfirmed,
+	})
+	must.Nil(t, err)
+	if innerTxn, ok := txn.Result.Transaction.(map[string]interface{}); ok {
+		if message, ok := innerTxn["message"].(map[string]interface{}); ok {
+			if instructions, ok := message["instructions"].([]interface{}); ok {
+				if instructionOne, ok := instructions[0].(map[string]interface{}); ok {
+					if parsed, ok := instructionOne["parsed"].(map[string]interface{}); ok {
+						if info, ok := parsed["info"].(map[string]interface{}); ok {
+							t.Log("Verifying chain transaction mint, to, and from")
+							should.Equal(t, "AH86ZDiGbV1GSzqtJ6sgfUbXSXrGKKjju4Bs1Gm75AQq", info["mint"])
+							should.Equal(t, state.PaymentDetails.To, info["wallet"])
+							should.Equal(t, state.PaymentDetails.From, info["source"])
+						} else {
+							t.Fail()
+						}
+					} else {
+						t.Fail()
+					}
+				} else {
+					t.Fail()
+				}
+				if instructionTwo, ok := instructions[1].(map[string]interface{}); ok {
+					if parsed, ok := instructionTwo["parsed"].(map[string]interface{}); ok {
+						if info, ok := parsed["info"].(map[string]interface{}); ok {
+							t.Log("Verifying chain transaction amount, ATA, and from")
+							amount := state.PaymentDetails.Amount.Mul(
+								decimal.NewFromFloat(math.Pow10(int(8))),
+							).BigInt().Uint64()
+							should.Equal(t, fmt.Sprint(amount), info["amount"])
+							should.Equal(t, ata.ToBase58(), info["destination"])
+							should.Equal(t, state.PaymentDetails.From, info["authority"])
+						} else {
+							t.Fail()
+						}
+					} else {
+						t.Fail()
+					}
+				} else {
+					t.Fail()
+				}
+			} else {
+				t.Fail()
+			}
+		} else {
+			t.Fail()
+		}
+	} else {
+		t.Fail()
+	}
```

Comment:
Sounds good - can you put a TODO comment in here to look at this later?

Reply:
Random passerby who happened to see this, it might've been preferable to just let it panic (thereby failing the test) instead of calling `t.Fail` by hand.